### PR TITLE
feat: Add DropDownButton

### LINKED ISSUE #2715: Support for DropDownButton
## What would you like to be added:

`DropDownButton`

## Why is this needed:

Feature parity

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #2715 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`DropDownButton` not supported.

## What is the new behavior?

`DropDownButton` supported

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using System.Threading.Tasks;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class DropDownButton : Button
+	{
+		public DropDownButton()
+		{
+			DefaultStyleKey = typeof(PersonPicture);
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs]: Invalid type

File: src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new DropDownButtonAutomationPeer(this);
+		}
+
+		//  - Missing
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs]: Why is it missing ?

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    x:Name="DropDownButtonTestPage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:MUXControlsTestApp"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:controls="using:Microsoft.UI.Xaml.Controls"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d">
+
+	<!--<Page.Resources>
```
Comment (agneszitte-nventive) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml]: Needed ?

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+{
+
+	[SampleControlInfo("DropDownButton", "MUX Page")]
+	public sealed partial class DropDownButtonPage
+	{
+		private int _clickCount = 0;
+		private int _flyoutOpenedCount = 0;
+		private int _flyoutClosedCount = 0;
+
+		private Flyout _flyout;
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml.cs]: @jeromelaban Should I address these or keep it (as it is direct copy from Windows UI tests?)

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml (hunk 3 above)
Comment (MartinZikmund) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml]: Probably not, copy from WinUI, but they don't use it either. Will remove it

File: src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs]: 😀 Ups

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml (hunk 3 above)
Comment (agneszitte-nventive) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml]: Thanks ;) !

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (carldebilly) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/DropDownButton/DropDownButtonPage.xaml.cs]: @MartinZikmund In this case you can ignore. The main goal of codacy is to find potential bugs...

File: src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/DropDownButton/DropDownButton.cs]: The code uses revokers - translating this to C# I would do subscribe and unsubscribe events - but the code will differ a bit (I will need to subscribe to `PropertyChanged` and use `OldValue`/`NewValue`. Is that ok or have you solved a similar situation differently?

**Pull request merged**